parlai/mturk/tasks/acute_eval/run.py

```diff
@@ -112,341 +102,379 @@ def add_args(from_argv=False):
+        """
+        onboarding_todo = list(range(len(self.onboarding_tasks)))
+        random.shuffle(onboarding_todo)
+        return self.worker_data.get(
+            worker_id,
+            {
+                'tasks_completed': [],
+                'conversations_seen': [],
+                'onboarding_todo': onboarding_todo,
+            },
+        )
 
-        task_group_id = mturk_manager.task_group_id
-        if opt['block_on_onboarding_fail'] and opt['block_qualification'] is None:
-            opt['block_qualification'] = task_group_id
+    def _supplement_opt(self):
+        """
+        Add additional args to opt.
+
+        Useful to add relevant options after args are parsed.
+        """
+        self.opt.update(
+            {
+                'task': os.path.basename(os.path.dirname(os.path.abspath(__file__))),
+                'task_description': {
+                    'num_subtasks': self.opt['subtasks_per_hit'],
+                    'question': self.opt['question'],
+                },
+                'frontend_version': 1,
+            }
+        )
+        self.opt.update(self.opt['task_config'])
+
+    def set_block_qual(self, task_group_id: str):
+        """
+        Set block qualification if necessary.
+
+        :param task_group_id:
+            task id used to set block qualification, if necessary.
+        """
+        if (
+            self.opt['block_on_onboarding_fail']
+            and self.opt['block_qualification'] is None
+        ):
+            self.opt['block_qualification'] = task_group_id
             warn_once(
                 "No block_qualification set in opt, automatically creating "
                 "new qualification {}".format(task_group_id)
             )
 
-        # Set up the sockets and threads to receive workers
-        mturk_manager.ready_to_accept_workers()
-
-        # Create the hits as specified by command line arguments
-        mturk_manager.create_hits()
-
-        def check_worker_eligibility(worker):
-            return True
-
-        def assign_worker_roles(workers):
-            workers[0].id = display_agent_name
-
-        def run_conversation(mturk_manager, opt, workers):
-            task_data = get_new_task_data(workers[0], opt['subtasks_per_hit'])
-            world = StaticMTurkTaskWorld(
-                opt, mturk_agent=workers[0], task_data=task_data
+    def _load_conversation_data(self):
+        """
+        Load conversation data.
+
+        Loads in the data from the pairs filepath.
+        """
+        pairs_path = self.opt.get('pairings_filepath')
+        if not os.path.exists(pairs_path):
+            raise RuntimeError('You MUST specify a valid pairings filepath')
+
+        with open(pairs_path) as pf:
+            for i, l in enumerate(pf.readlines()):
+                convo_pair = json.loads(l.strip())
+                task = {
+                    'task_specs': {
+                        'conversation_order': random.choice([[0, 1], [1, 0]]),
```

Comment:
we can improve this. A better option is to have a key for model_left and a key for model_right, and then we should dump to disk the name of the winner, i.e. the value inside `model_left` or the value for `model_b`. This would greatly simplify the analysis logic.

Reply:
agreed, i'll look into a better way of handling this

Reply:
this will affect the REACT component as well just FYI

Reply:
i'm just documenting my thought process at this point but this `conversation_order` is not actually used in analysis and is purely for the javascript rendering, so i'm going to leave as is

Reply:
I'd already made the changes to analysis logic prior to this - I had it this way here to avoid dealing with names in this script, and I would rather have this than have rematch strings for display.

Reply:
Why can't we use model_left and model_right, and update the react component?